src/ring_theory/jacobson.lean

```diff
@@ -252,4 +259,156 @@ end
 
 end localization
 
+section polynomial
+open polynomial
+
+/-- If `f : R → S` descends to an integral map in the localization at `x`,
+  and `R` is a jacobson ring, then the intersection of all maximal ideals in `S` is empty -/
+lemma jacobson_bot_of_integral_localization {R S : Type*} [integral_domain R] [integral_domain S]
+  {Rₘ Sₘ : Type*} [comm_ring Rₘ] [comm_ring Sₘ] [is_jacobson R]
+  (φ : R →+* S) (hφ : function.injective φ) (x : R) (hx : x ≠ 0)
+  (ϕ : localization_map (submonoid.powers x) Rₘ)
+  (ϕ' : localization_map ((submonoid.powers x).map φ : submonoid S) Sₘ)
+  (hφ' : (ϕ.map ((submonoid.powers x).mem_map_of_mem (φ : R →* S)) ϕ').is_integral) :
+  (⊥ : ideal S).jacobson = ⊥ :=
+begin
+  have hM : ((submonoid.powers x).map φ : submonoid S) ≤ non_zero_divisors S :=
+    map_le_non_zero_divisors_of_injective hφ
+      (le_non_zero_divisors_of_domain (λ h0, hx (let ⟨n, hn⟩ := h0 in pow_eq_zero hn))),
+  letI : integral_domain Sₘ := localization_map.integral_domain_of_le_non_zero_divisors ϕ' hM,
+  let φ' : Rₘ →+* Sₘ := ϕ.map ((submonoid.powers x).mem_map_of_mem (φ : R →* S)) ϕ',
+  suffices : ∀ I : ideal Sₘ, I.is_maximal → (I.comap ϕ'.to_map).is_maximal,
+  { have hϕ' : comap ϕ'.to_map ⊥ = ⊥ :=
+    by simpa [ring_hom.injective_iff_ker_eq_bot, ring_hom.ker_eq_comap_bot] using (ϕ'.injective hM),
+    refine eq_bot_iff.2 (le_trans _ (le_of_eq hϕ')),
+    have hSₘ : is_jacobson Sₘ := is_jacobson_of_is_integral' φ' hφ' (is_jacobson_localization ϕ),
+    rw [← hSₘ ⊥ radical_bot_of_integral_domain, comap_jacobson],
+    exact Inf_le_Inf (λ j hj, ⟨bot_le, let ⟨J, hJ⟩ := hj in hJ.2 ▸ this J hJ.1.2⟩) },
+  introsI I hI,
+  -- Remainder of the proof is pulling and pushing ideals around the square and the quotient square
```

Comment:
The rest of this proof is just pulling the ideals around the square to prove the statement of the `suffices` just above. I'm not sure if that is really a generally useful thing, so I'm not sure whether it would make sense to pull out into a new theorem?